artifactory/commands/golang/archive.go

```diff
@@ -134,8 +119,13 @@ func archiveProject(writer io.Writer, dir, mod, version string) error {
 				return filepath.SkipDir
 			}
 		}
+
 		if info.Mode().IsRegular() {
-			if !isVendoredPackage(slashPath) {
+			excluded, err := isPathExcluded(filePath, excludePathPattern)
+			if err != nil {
+				return err
+			}
+			if !isVendoredPackage(slashPath) && !excluded {
 				files = append(files, dirFile{
 					filePath:  filePath,
 					slashPath: slashPath,
```

Comment:
The above code can optimized as follows, to avoid calling the `isPathExcluded` in some cases - 
```
		if info.Mode().IsRegular() {
			if !isVendoredPackage(slashPath) {
				excluded, err := isPathExcluded(filePath, excludePathPattern)
				if err != nil {
					return err
				}
				if !excluded {
					files = append(files, dirFile{
						filePath:  filePath,
						slashPath: slashPath,
						info:      info,
					})
				}
			}
			return nil
		}
```